app/assets/javascripts/govuk_publishing_components/lib/cookie-functions.js

```diff
@@ -114,12 +129,29 @@
       window.GOVUK.setDefaultConsentCookie()
     }
 
-    if (COOKIE_CATEGORIES[cookieName]) {
-      var consent = JSON.parse(window.GOVUK.getCookie('cookie_policy'))
+    currentConsentCookie = window.GOVUK.getConsentCookie()
 
+    // Survey cookies are tested for in Static, which we need to allow
+    if (cookieName.match('^govuk_surveySeen') || cookieName.match('^govuk_taken')) {
+      // Sometimes currentConsentCookie is malformed in some of the tests, so we need to handle these
```

Comment:
Could we somehow move this check into `getConsentCookie`? Or perhaps querying the consent for a category, e.g: 'essential' could be a function in itself. Otherwise we have code duplication every time we want to check the cookie.